src/java/org/apache/cassandra/db/guardrails/GuardrailsConfigService.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.db.guardrails;
+
+import java.util.Set;
+
+import com.google.common.collect.ImmutableSet;
+
+import org.apache.cassandra.config.DatabaseDescriptor;
+import org.apache.cassandra.config.GuardrailsOptions;
+import org.apache.cassandra.utils.MBeanWrapper;
+
+/**
+ * Implementation of {@link GuardrailsConfigServiceMBean}.
+ */
+public class GuardrailsConfigService implements GuardrailsConfigServiceMBean
+{
+    public static final String MBEAN_NAME = "org.apache.cassandra.db:type=Guardrails";
+
+    private final GuardrailsOptions config = DatabaseDescriptor.getGuardrailsConfig();
+
+    GuardrailsConfigService()
+    {
+        MBeanWrapper.instance.registerMBean(this, MBEAN_NAME);
+    }
+
+    @Override
+    public boolean getEnabled()
+    {
+        return config.getEnabled();
+    }
+
+    @Override
+    public void setEnabled(boolean enabled)
+    {
+        config.setEnabled(enabled);
+    }
+
+    @Override
+    public int getTablesWarnThreshold()
+    {
+        return (int) config.getTables().getWarnThreshold();
```

Comment:
nit: In TrackWarnings we just relied on a different type for int and long; that way casting isn't needed and there is no confusion about min/max values

Reply:
I don't think that casting on JMX methods is going to be too harmful. The configs in `GuardrailOptions` are separated by types, but I would keep `Threshold` and `Threshold.Config` based on longs for simplicity, to avoid code duplication nor playing with generics.

Reply:
Indeed, I'm modifying `GuardrailOptions.Threshold` to always use longs, relying on validation to verify that the values are in the int range when needed. I think that makes all the threshold stuff easier to maintain and the performance cost of upcasting when checking the guardrail will be negligible. 

What would be the alternative? Generics or just duplicating the `Threshold` class and their configs? I see that `track_warning` duplicates the config and it's validation, but in the case of guardrails the duplication of `Threshold` and it's config seems more problematic just because it is bigger, and it will be even bigger if we add counters, diagnostic events, etc. to it. As for generics, I think we would be trading widening for autoboxing.